crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,380 @@
+    #[inline]
+    pub fn degrees(degrees: f32) -> Self {
+        Self::radians(degrees.to_radians())
+    }
+
+    /// Creates a [`Rotation2d`] from the sine and cosine of an angle in radians.
+    ///
+    /// The rotation is only valid if `sin.asin() == cos.acos()`.
+    #[inline]
+    pub const fn from_sin_cos(sin: f32, cos: f32) -> Self {
+        Self { sin, cos }
+    }
+
+    /// Returns the rotation in radians in the `(-pi, pi]` range.
+    #[inline]
+    pub fn as_radians(self) -> f32 {
+        #[cfg(feature = "libm")]
+        {
+            libm::atan2(self.sin as f64, self.cos as f64) as f32
+        }
+        #[cfg(not(feature = "libm"))]
+        {
+            f32::atan2(self.sin, self.cos)
+        }
+    }
+
+    /// Returns the rotation in degrees in the `(-180, 180]` range.
+    #[inline]
+    pub fn as_degrees(self) -> f32 {
+        self.as_radians().to_degrees()
+    }
+
+    /// Returns the sine and cosine of the rotation angle in radians.
+    #[inline]
+    pub const fn sin_cos(self) -> (f32, f32) {
+        (self.sin, self.cos)
+    }
+
+    /// Returns `true` if the rotation is neither infinite nor NaN.
+    #[inline]
+    pub fn is_finite(self) -> bool {
+        self.sin.is_finite() && self.cos.is_finite()
+    }
+
+    /// Returns `true` if the rotation is NaN.
+    #[inline]
+    pub fn is_nan(self) -> bool {
+        self.sin.is_nan() || self.cos.is_nan()
+    }
+
+    /// Returns `true` if the rotation is near [`Rotation2d::IDENTITY`].
+    #[inline]
+    pub fn is_near_identity(self) -> bool {
+        // TODO: We might be able to use the rotation sine and cosine
+        //       directly instead of first converting to radians.
+        // Same as `Quat::is_near_identity`
+        let threshold_angle = 0.002_847_144_6;
+        self.as_radians().abs() < threshold_angle
+    }
+
+    /// Returns the angle in radians needed to make `self` and `other` coincide.
+    #[inline]
+    pub fn angle_between(self, other: Self) -> f32 {
+        (other * self.inverse()).as_radians()
+    }
+
+    /// Returns the inverse of the rotation. This is also the conjugate
+    /// of the unit complex number representing the rotation.
+    #[inline]
+    #[must_use]
+    #[doc(alias = "conjugate")]
+    pub fn inverse(self) -> Self {
+        Self {
+            cos: self.cos,
+            sin: -self.sin,
+        }
+    }
+
+    /// Performs a linear interpolation between `self` and `rhs` based on
+    /// the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s`
+    /// is `1.0`, the result will be equal to `rhs`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Rotation2d;
+    /// #
+    /// let rot1 = Rotation2d::IDENTITY;
+    /// let rot2 = Rotation2d::radians(std::f32::consts::FRAC_PI_2);
+    ///
+    /// let result = rot1.lerp(rot2, 0.5);
+    ///
+    /// assert_eq!(result.as_radians(), std::f32::consts::FRAC_PI_4);
+    /// ```
```

Comment:
Personally I think that the docs for `lerp` and `slerp` could highlight the difference between the two a bit more. 

The current example for `lerp` doesn't really help since the `a=0.5` value might be cool to get some well-known results ($\frac{\pi}{4}$), but on the other hand `slerp`-ing with the same `a` value would yield the same result. 

Here are some ideas how we could improve the situation:

- for both `lerp` and `slerp` use the same input rotations and `a` for a second example. Maybe make this example a bit more "extreme" where the angle between the input and output is close to $\pi$
- a small section listing some of the use cases for each method to further inform the user what this is for

Reply:
I made the examples use some slightly more interesting values, and added more tests for lerp and slerp.

I also improved the docs a bit, but I'm not quite sure how to describe lerp nicely; it kinda gives an ease-in-out effect, but the closer the angles are to being opposite (like 0 and pi), the more "aggressive" the interpolation is, where it essentially snaps from the start to the end almost immediately. I also tried explaining what it does mathematically, but I'm not 100% sure if it's entirely correct

Reply:
Looks better now! Thank you for spending a bit more time on it!